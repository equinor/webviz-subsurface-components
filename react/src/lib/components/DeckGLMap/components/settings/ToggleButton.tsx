import React, { ChangeEvent } from "react";
import { Label, Switch } from "@equinor/eds-core-react";

interface Props {
    /**
     * Label for the component.
     */
    label: string;
    /**
     * Initial state of the component.
     */
    checked: boolean;
    /**
     * Callback to update the state of the component.
     */
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const ToggleButton: React.FC<Props> = React.memo(
    ({ label, checked, onChange }: Props) => {
        return (
            <div
                style={{
                    display: "flex",
                    justifyContent: "space-between",
                }}
            >
                <Label
                    label={label}
                    id={`${label}-switch-label`}
                    style={{
                        paddingTop: 15,
                        fontSize: 15,
                    }}
                />
                <Switch
                    id={`${label}-switch`}
                    aria-label={label}
                    label={""}
                    onChange={onChange}
                    checked={checked}
                    style={{
                        paddingRight: 10,
                        width: "3rem",
                    }}
                />
            </div>
        );
    }
);

ToggleButton.displayName = "ToggleButton";
export default ToggleButton;
