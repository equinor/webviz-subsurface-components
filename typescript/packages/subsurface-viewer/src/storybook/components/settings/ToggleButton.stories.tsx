import type { Meta, StoryObj } from "@storybook/react";

import ToggleButton from "../../../components/settings/ToggleButton";

const stories: Meta = {
    component: ToggleButton,
    title: "SubsurfaceViewer/Components/Settings",
};
export default stories;

export const ToggleButtonStory: StoryObj<typeof ToggleButton> = {
    name: " ToggleButton",
    args: {
        label: "test",
    },
};
