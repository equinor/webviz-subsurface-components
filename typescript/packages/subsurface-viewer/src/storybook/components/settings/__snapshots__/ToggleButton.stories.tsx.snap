// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubsurfaceViewer/Components/Settings ToggleButtonStory smoke-test: subsurfaceviewer-components-settings--toggle-button-story.dom.html 1`] = `
<div style="display: flex; justify-content: space-between;">
  <label id="test-switch-label"
         class="Label__LabelBase-sc-1gi2bcn-0 fdTexl"
         style="padding-top: 15px; font-size: 15px;"
  >
    <span class="Label__Text-sc-1gi2bcn-1 hkGcWj">
      test
    </span>
  </label>
  <span class="Switchstyles__GridWrapper-sc-x39lde-2 gUdGex"
        style="padding-right: 10px; width: 3rem;"
  >
    <input id="test-switch"
           aria-label="test"
           type="checkbox"
           class="Switchstyles__BaseInput-sc-x39lde-1 SwitchDefault__Input-sc-16ym5pn-0 kNKTpL ewbgpJ"
    >
    <span class="Switchstyles__BaseInputWrapper-sc-x39lde-0 fSSKMh">
      <span class="SwitchDefault__Track-sc-16ym5pn-1 kPsJrV">
      </span>
      <span class="SwitchDefault__Handle-sc-16ym5pn-2 jdwCBx">
      </span>
    </span>
  </span>
</div>
`;
